src/d3a/models/strategy/external_strategies/__init__.py

```diff
@@ -284,11 +284,13 @@ def _publish_trade_event(self, trade, is_bid_trade):
                                    "local_market_fee":
                                        self.area.current_market.fee_class.grid_fee_rate
                                        if self.area.current_market is not None else "None",
-                                   "attributes": {},
+                                   "attributes": trade.offer_bid.attributes,
                                    "seller": trade.seller
                                    if trade.seller_id == self.device.uuid else "anonymous",
                                    "buyer": trade.buyer
                                    if trade.buyer_id == self.device.uuid else "anonymous",
+                                   "seller_origin": trade.seller_origin,
+                                   "buyer_origin": trade.buyer_origin,
```

Comment:
I remember that we had a number of issues with the missing of these parameters. I wonder why there are still places were we have to re-add these.